File: careerforge-frontend/src/services/chatbotService.ts
import { apiClient } from '../config/api';

export interface ChatMessage {
  role: 'user' | 'assistant';
  content: string;
  isOfflineKnowledge?: boolean;
}

export interface ChatbotResourceLink {
  title: string;
  url: string;
  type: 'ROADMAP' | 'SHEET' | 'RESOURCE' | 'EXTERNAL';
}

export interface ChatbotResponse {
  answer: string;
  queriesRemaining: number;
  dailyLimit: number;
  isOfflineKnowledge: boolean;
  suggestedFollowUps?: string[];
  relevantLinks?: ChatbotResourceLink[];
}

export interface ChatbotQuota {
  dailyLimit: number;
  queriesUsedToday: number;
  queriesRemaining: number;
  cooldownSecondsRemaining: number;
  resetsAt: string;
}

export interface ChatbotRequest {
  message: string;
  history?: { role: string; content: string }[];
  pageContext?: string;
}

export const chatbotService = {
  async askQuestion(request: ChatbotRequest): Promise<ChatbotResponse> {
    const response = await apiClient.post('/api/student/chatbot/ask', request);
    return response.data.data;
  },

  async getQuota(): Promise<ChatbotQuota> {
    const response = await apiClient.get('/api/student/chatbot/quota');
    return response.data.data;
  },

  async getSuggestedPrompts(): Promise<string[]> {
    const response = await apiClient.get('/api/student/chatbot/suggested-prompts');
    return response.data.data;
  }
};

File: careerforge/src/main/java/com/careerforge/chatbot/service/ChatbotService.java
package com.careerforge.chatbot.service;

import com.careerforge.chatbot.dto.ChatbotQuotaDto;
import com.careerforge.chatbot.dto.ChatbotRequest;
import com.careerforge.chatbot.dto.ChatbotResponse;

import java.util.List;
import java.util.UUID;

public interface ChatbotService {

    ChatbotResponse askQuestion(UUID userId, ChatbotRequest request);

    ChatbotQuotaDto getStudentQuota(UUID userId);

    List<String> getSuggestedPrompts(UUID userId);
}
